WooCommerce/Classes/ViewRelated/Products/ProductSearchViewController.swift

```diff
@@ -0,0 +1,506 @@
+import UIKit
+import Yosemite
+
+/// ProductSearchViewController: Displays the "Search Products" Interface
+///
+class ProductSearchViewController: UIViewController {
```

Comment:
This VC seems pretty similar, as you mentioned, to OrderSearchViewController. Have you considered reusing it? 

I think it would make sense to explore transforming this into a generic SearchViewController, generic on the model object to be rendered? The UI is basically the same, and the only thing that changes is the cell (which could most likely be made generic as well). That coupled with a more abstract transition coordinator (as mentioned in a previous PR) could help remove a significant amount of boilerplate

Reply:
Seconded. I'd love to see a generic search view controller. (The transition coordinator could be broken out into its own file as well.)